plugins/tekton/src/hooks/usePodLogs.ts

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import useAsyncRetry from 'react-use/lib/useAsyncRetry';
+import useInterval from 'react-use/lib/useInterval';
+
+import { useApi } from '@backstage/core-plugin-api';
+import { kubernetesProxyApiRef } from '@backstage/plugin-kubernetes';
+
+import { V1Container, V1Pod } from '@kubernetes/client-node';
+
+import { TektonResourcesContextData } from '../types/types';
+import { TektonResourcesContext } from './TektonResourcesContext';
+
+export interface ContainerScope {
+  podName: string;
+  podNamespace: string;
+  clusterName: string;
+  containerName: string;
+}
+
+interface PodLogsOptions {
+  pod: V1Pod;
+  intervalMs?: number;
+}
+
+export const usePodLogs = ({ pod, intervalMs = 5000 }: PodLogsOptions) => {
```

Comment:
we should have a differentiation in name if we can't have one common hook I think

check `usePodLogs` in topology plugin

Reply:
The `usePodLogs` in the Topology plugin fetch the logs of a container
but the hook in the Tekton Plugin fetches the logs from all the containers of a Pod

I have updated the name of this hook to `usePodLogsOfPipelineRun`